snap/validate.go

```diff
@@ -183,6 +280,21 @@ func ValidateApp(app *AppInfo) error {
 			return err
 		}
 	}
+
+	// Socket activation requires the "network-bind" plug
```

Comment:
is this a request from security, our own decision wrt policy, or ...?

Reply:
It came from discussions (not a request from security). I think it makes sense, if an app can't bind the network it shouldn't be able to be socket-activated

Reply:
To be clear, the issue isn't that it shouldn't be able to. The issue is that it _cannot,_ right?

Reply:
it should be able to: the one doing the binding is systemd, which is not confined by us

Reply:
Yes, but there are other socket operations that will be performed in the handed-off file descriptor I suppose? Do those generally overlap with the set defined in the network-bind interface?

Another question is the intent of the plug. Right now one can tell whether a snap will be listening for traffic in the local machine based on the plugs. If we don't force that plug to be present, snaps will be able to listen without an obvious hint or the ability to prevent that. That said, the mere presence of the plug in the snap isn't enough to make it conform. We'd need to force it to be _connected_ as well for the activation to take place.

This might be an argument to drop the plug and just let it go. On the other hand, if we don't force its presence today, we'll not be able to change exactly that behavior later, because we'd break compatibility. If we force the plug to be present, we can change that behavior and allow auto-connection to take place, and we'd know for sure that any snaps that ever existed with socket activation will necessarily continue working.

So, on that basis, I'd prefer to merge the code with the enforcement of the plug present as it is now, and if we change our minds later we can drop that logic.